include/aspect/particle/generator/probability_density_function.h

```diff
@@ -112,6 +112,14 @@ namespace aspect
           types::particle_index n_particles;
 
           /**
+           * If true, particle numbers per cell are calculated randomly
+           * according to their respective probability density. If false,
+           * particle numbers per cell follow exactly the particle density,
+           * only the locations of particles are chosen randomly.
+           */
+          bool random_cell_selection;
```

Comment:
I don't think I understand the comment. Can you document this in terms of the algorithm behind it, e.g. "first determine how many particles each cell should have based on the integral of the density over each of the cells, and then once we know how many particles we want on each cell, choose their locations randomly within each cell."